src/lib/datepicker/datepicker-input.ts

```diff
@@ -285,6 +286,7 @@ export class MdDatepickerInput<D> implements AfterContentInit, ControlValueAcces
     let date = this._dateAdapter.parse(value, this._dateFormats.parse.dateInput);
     this._lastValueValid = !date || this._dateAdapter.isValid(date);
     date = this._getValidDateOrNull(date);
+    this._value = date;
```

Comment:
Why not just go through the setter? A lot of this logic looks the same. I think you could change this method to:

```ts
let date = this._dateAdapter.parse(value, this._dateFormats.parse.dateInput);
this.value = date;
this._cvaOnChange(this.date);
this.dateInput.emit(new MdDatepickerInputEvent(this, this._elementRef.nativeElement));
```

Reply:
@mmalerba `this.value = date` can't be used in `_onInput()` because the setter of `value` formats the date and sets it as the property on the native element so simply typing in 1 on immediately trigger update and set the value to 1.1.2001.

`this._renderer.setProperty(this._elementRef.nativeElement, 'value', value ? this._dateAdapter.format(value, this._dateFormats.display.dateInput) : '');`

We would only want the `DateAdapter.format` update when the date is chosen by clicking on a date in datepicker.

Reply:
Oh I see right, that makes sense